src/ssh/cockpitsshrelay.c

```diff
@@ -667,12 +634,11 @@ set_knownhosts_file (CockpitSshData *data,
               if (!res)
                 goto out;
 
-              /* this should always be known now, but let's make double-sure */
-              host_known = cockpit_is_host_known (tmp_knownhost_file, host, port);
+              host_known = session_has_known_host_in_file (tmp_knownhost_file, data, host, port);
               if (host_known)
                 data->ssh_options->knownhosts_file = tmp_knownhost_file;
               else
-                g_warning ("sss_ssh_knownhostsproxy reported key for %s:%u which is not known to cockpit_is_host_known()", host, port);
+                g_debug ("sss_ssh_knownhostsproxy reported key for %s:%u which is not known to session_has_known_host_in_file()", host, port);
```

Comment:
Yup, which is why I temporarily changed it, as the sssd_badkey one does fail. 

The libssh api just recognizes it as unknown at this stage.

However the old cockpit api for some reason doesn't see the bad key as unknown, but as known; proceeds to connect but fails because the key is invalid.

I think this is one of the only instances where code path actually changes, and in this case I think it's totally fine/makes more sense.